### ISSUE TITLE
TypeError on empty response from HTTP request

### ISSUE THREAD

**Issue by hieronx** [jokade/scalajs-angulate#71]
I've got the following `Service` for retrieving some data from our API:

``` scala
class UserService($http: HttpService) extends Service {
   def findMe(): HttpPromise[User] = $http.get(s"$host/users/me")
}
```

And I'm calling this using the following code:

``` scala
54:   userService.findMe() onComplete {
55:      case Success(data) => js.Dynamic.global.console.log(data)
56:      case Failure(ex) => js.Dynamic.global.console.error(s"An error has occurred: $ex")
57:   }
```

Now, when `/users/me` returns a 200 OK with the correct data, it works perfectly. However, when it returns either a `404` or `405` without any response data, the following error occurs instead of triggering the `Failure` case:

```
TypeError: Cannot read property 'toString' of null
    at $objectToString (scalajsenv.js:192)
    at App.scala:54
    [..]
```

`App.scala:54` returns to the onComplete method in the 2nd code block. This seems somewhat similar to #22, yet I'm using the latest version (`0.2.2-SNAPSHOT`), so it should already be fixed. Any thoughts on what's going on here?

**Comment by jokade** [comment]
Does this error also occur, when you use `onSuccess` and `onFailure` instead of `onComplete`?

**Comment by hieronx** [comment]
Same problem, I've tried the following:

``` scala
userService.findMe() onSuccess {
   case data => js.Dynamic.global.console.log(data)
}

userService.findMe() onFailure {
   case ex => js.Dynamic.global.console.error(s"An error has occurred: $ex")
}
```

This creates the same error message:

```
TypeError: Cannot read property 'toString' of null
    at $objectToString (scalajsenv.js:192)
    at App.scala:61
```

`App.scala:61` returns to the first line of the code block.

**Comment by jokade** [comment]
Could you please check what happens if you use the bare bone Angular method `error()` instead of `onFailure`?

**Comment by hieronx** [comment]
Yeah, the following code works just like it should:

``` scala
userService.findMe().success((data: String, status: Int) =>
   js.Dynamic.global.console.log(data)
)

userService.findMe().error((data: String, status: Int) =>
   js.Dynamic.global.console.error(data)
)
```

**Comment by jokade** [comment]
Thanks; I have to check the macro code, but that will take some time (I'm currently preoccupied with other stuff)...

**Comment by hieronx** [comment]
Of course, I can just use the plain angular promise methods for now. Let me know if you need any more debugging from my end later on.

**Comment by jokade** [comment]
@offerijns I think I've solved the issue in the current 0.2.2-SNAPSHOT; could you please verify?

**Comment by hieronx** [comment]
@jokade yes it's fixed, thanks a lot!
